probe/probe.go

```diff
@@ -360,119 +379,197 @@ type KubeArmorProbeData struct {
 	HostVisibility          string
 }
 
-func isKubeArmorRunning(c *k8s.Client, o Options) bool {
-	_, err := getKubeArmorDaemonset(c, o)
-	return err == nil
+func isKubeArmorRunning(c *k8s.Client, o Options) (bool, *Status) {
+	isRunning, DaemonsetStatus := getKubeArmorDaemonset(c, o)
+	return isRunning, DaemonsetStatus
+
+}
 
+// Status data
+type Status struct {
+	Desired   string `json:"desired"`
+	Ready     string `json:"ready"`
+	Available string `json:"available"`
 }
 
-func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, error) {
+func getKubeArmorDaemonset(c *k8s.Client, o Options) (bool, *Status) {
 	var data [][]string
 	// KubeArmor DaemonSet
 	w, err := c.K8sClientset.AppsV1().DaemonSets(o.Namespace).Get(context.Background(), "kubearmor", metav1.GetOptions{})
 	if err != nil {
-		return false, err
+		return false, nil
```

Comment:
why are we sending nil if there is an error? shouldn't we at least log this error?

Reply:
done